File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
    let left = 0
    let right = nums.length - 1
    let index = 0
    while (index <= right) {
        if (nums[index] == 0) {
            [nums[index], nums[left]] = [nums[left], nums[index]]
            left++
            index++
        }
        else if (nums[index] == 2) {
            [nums[index], nums[right]] = [nums[right], nums[index]]
            right = right - 1
        }
        else {
            index++
        }
    }
    return nums
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        lt = 0
        gt = len(nums)-1
        i = 0
        pivot = 1

        while i <= gt:
            if nums[i] < pivot:
                nums[i],nums[lt] = nums[lt],nums[i]
                lt+=1;
                i+=1;
            elif nums[i] > pivot:
                nums[i],nums[gt]=nums[gt],nums[i]
                gt-=1;
            else: 
                i+=1;
        